primefaces/src/main/resources/META-INF/resources/primefaces/core/core.widget.js

```diff
@@ -393,6 +394,17 @@ if (!PrimeFaces.widget) {
                 }
             }
             this.destroyListeners = [];
+
+            // remove any event listeners on jQuery objects
```

Comment:
please a bit more docu here
which contains that we loop all stored vars of the current widget, and unbding all event listeners to avoid mem leaks

also shouldnt we do addtionally:
this[key] = null 
just to be sure?

Reply:
I will update the docs to be more clear.  

I was originally doing `this[key] =null` here as well but some components failed in integration tests and my client had multiple issues with components where if there was a "delay" it was trying to use the `jq` and blowing up.  So I removed it and so far it has not affected any garbage collection.

For example here is a patch i had to add that was failing with `this[key]=null;` here so i had to defensively check the jq but it was not worth it to find it all over the codebase.  In my client that has the severe memory leak everything is garbage collecting fine without `this[key] =null` so I have chosen to leave it out and continue testing!

```js
/**
 * TieredMenu fix is a defensive NPE check on jq after our memory cleanup in destroyDetachedWidgets.
 */
if (PrimeFaces.widget.TieredMenu) {
  PrimeFaces.widget.TieredMenu.prototype.reset = function () {
    var $this = this;
    this.active = false;

    if (this.jq) {
      this.jq.find("li.ui-menuitem-active").each(function () {
        $this.deactivate($(this), true);
      });
    }
  };
}
```

Reply:
comments and docs added